ruby/private/sdk.bzl

```diff
@@ -1,50 +1,48 @@
-load(
-    "@rules_ruby//ruby/private/toolchains:ruby_runtime.bzl",
-    _ruby_runtime = "ruby_runtime",
-)
+load("@rules_ruby//ruby/private/toolchains:ruby_runtime.bzl", "ruby_runtime")
+load (":constants.bzl", "SUPPORTED_VERSIONS")
 
-def rules_ruby_select_sdk(version = "host"):
+def _register_toolchain(version):
     """Registers ruby toolchains in the WORKSPACE file."""
+    name = "local_config_ruby_%s" % version
+    supported_version = None
 
-    supported_versions = [
-        "host",
-        "2.5.8",
-        "2.5.9",
-        "2.6.3",
-        "2.6.4",
-        "2.6.5",
-        "2.6.6",
-        "2.6.7",
-        "2.6.8",
-        "2.6.9",
-        "2.7.1",
-        "2.7.2",
-        "2.7.3",
-        "2.7.4",
-        "2.7.5",
-        "3.0.0",
-        "3.0.1",
-        "3.0.2",
-        "3.0.3",
-        "3.1.0",
-        "3.1.1",
-        "jruby-9.2.21.0",
-        "jruby-9.3.6.0",
-    ]
-
-    for v in sorted(supported_versions, reverse=True):
+    version = version.removeprefix("ruby-")
+    for v in sorted(SUPPORTED_VERSIONS, reverse=True):
         if v.startswith(version):
             supported_version = v
             break
 
     if not supported_version:
-        fail("rules_ruby_select_sdk: unsupported ruby version '%s' not in '%s'" % (version, supported_versions))
+        fail("rules_ruby_register_toolchains: unsupported ruby version '%s' not in '%s'" % (version, SUPPORTED_VERSIONS))
 
-    _ruby_runtime(
-        name = "org_ruby_lang_ruby_toolchain",
+    ruby_runtime(
+        name = name,
         version = supported_version,
     )
 
     native.register_toolchains(
-        "@org_ruby_lang_ruby_toolchain//:toolchain",
+        "@%s//:toolchain" % name,
+    )
+
+def rules_ruby_register_toolchains(version = None):
```

Comment:
The implicit registration of "system" and the magical meaning of `version = None` don't feel right to me. What about making it all explicit:

```
rules_ruby_register_toolchains(
    versions = [
        "system",
        "3.0.3",
    ]
)
```

For `3.0.3` vs `3.0` it seems that the string given here should correspond to whatever you would pass to `--@rules_ruby//ruby:version=xyz`

Reply:
Good point, I updated it to just take an explicit list of versions, which more closely match the flags.